dali/python/nvidia/dali/_multiproc/shared_batch.py

```diff
@@ -217,42 +192,60 @@ def _add_array_to_batch(self, np_array, meta, memview):
     def _write_batch(self, batch):
         if not batch:
             return
-        batch = [(idx, _apply_to_sample(lambda x: _sample_to_numpy(x, _sample_error_msg), sample))
-                 for idx, sample in batch]
+        batch = [_apply_to_sample(lambda x: _sample_to_numpy(x, _sample_error_msg), sample)
+                 for sample in batch]
         meta, data_size = self._prepare_samples_meta(batch)
         serialized_meta = pickle.dumps(meta)
         self.meta_data_size = len(serialized_meta)
         self.data_size = _align_up(data_size, self.SAMPLE_ALIGNMENT)
-        needed_capacity = self.data_size + self.meta_data_size
-        if self.mem_batch.capacity < needed_capacity:
-            new_capacity = max(needed_capacity, 2 * self.mem_batch.capacity)
-            new_capacity = _align_up(new_capacity, self.BUFFER_ALIGNMENT)
-            self.mem_batch.resize(new_capacity)
-        memview = self.mem_batch.shm_chunk.buf
-        for (idx, sample), (meta_idx, sample_meta) in zip(batch, meta):
-            assert idx == meta_idx
+        self.total_size = _align_up(self.data_size + self.meta_data_size, self.SAMPLE_ALIGNMENT)
+        if self.shm_chunk.capacity < self.total_size:
+            self.resize_shm_chunk(self.shm_chunk, self.total_size + self.min_trailing_offset)
```

Comment:
Similar to L168 - shouldn't it be aligned?

Reply:
I can make it explicitly aligned here, but resize_shm_chunk aligns that value up to the BUFFER_ALIGNMENT.

Reply:
No need in such case.